pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -844,6 +852,18 @@ func (npm *nominatedPodMap) UpdateNominatedPod(oldPod *v1.Pod, newPodInfo *frame
 // NewPodNominator creates a nominatedPodMap as a backing of framework.PodNominator.
 func NewPodNominator() framework.PodNominator {
 	return &nominatedPodMap{
+		podAlive:           func(_ *v1.Pod) bool { return true },
+		nominatedPods:      make(map[string][]*framework.PodInfo),
+		nominatedPodToNode: make(map[types.UID]string),
+	}
+}
+
+// NewSafePodNominator creates a nominatedPodMap as a backing of framework.PodNominator.
+// Unlike NewPodNominator, it passes in a function to check if the pod is alive before
+// adding its nominatedNode info.
+func NewSafePodNominator(podAlive func(pod *v1.Pod) bool) framework.PodNominator {
```

Comment:
can we make this accept the lister?

Reply:
As I mentioned in https://github.com/kubernetes/kubernetes/pull/102498#discussion_r645053765, I tend of pass in a function rather than restraining it to a podLister, so callers (tests or other integrators) can easily build this.

Reply:
Tests should be using the prod behavior when possible.